### ISSUE TITLE
Environment variables regressions

### ISSUE THREAD

**Issue by andreeis** [microsoft/vscode-cmake-tools#2354]
When using kits, any environment variable coming from the setup script or the "environmentVariables" in a kit are not merged into the final env.

Additionally, any environment variable defined in the terminal from which VSCode is launched is also lost and doesn't make it into the final supposedly complete list.

**Comment by bobbrow** [comment]
@lygstate after merging your PR, we ran into a few issues.  Would you be able to help address them?  @andreeis would you be able to help write some repro steps for this?

**Comment by lygstate** [comment]
> @lygstate after merging your PR, we ran into a few issues. Would you be able to help address them? @andreeis would you be able to help write some repro steps for this?
OK,  @andreeis can you provide a simple repository to reproduce the issue?

**Comment by andreeis** [comment]
With latest CMT (git pull the main branch as of now), which contains the env var PR, I wrote some temporary hacky code in environmentVariables.ts, merge method so that every time an env is merged into finalEnv I search for my own environment variables. I admit, I didn't create a repro that would be more natural and closer to a user scenario. Still, the code below is hit when presets are on and not hit when kits are on. And the environment variable defined in the terminal before I even open VSCode is not added in any of the two cases (kits or presets).

Hacky code in environmentVariables.ts:

```
    public static merge(envs: (EnvironmentWithNull | null | undefined)[], options?: EnvironmentOptions): Environment {
      const newEnv = EnvironmentUtils.create(undefined, options);
      const log = logging.createLogger('mytemp');
      for (const env of envs) {
          if (env !== undefined && env !== null) {
              Object.assign(newEnv, env);

              if (env["MyEnv_Variable_Kit"] !== "" && env["MyEnv_Variable_Kit"] !== undefined) {
                 log.warning(`ENV["MyEnv_Variable_Kit"] = "${env["MyEnv_Variable_Kit"]}"`);
              }

              if (env["MyEnv_Variable_Script"] !== "" && env["MyEnv_Variable_Script"] !== undefined) {
                log.warning(`ENV["MyEnv_Variable_Script"] = "${env["MyEnv_Variable_Script"]}"`);
             }

             if (env["MyEnv_Variable_Prompt"] !== "" && env["MyEnv_Variable_Prompt"] !== undefined) {
                log.warning(`ENV["MyEnv_Variable_Prompt"] = "${env["MyEnv_Variable_Prompt"]}"`);
             }

             if (env["MyEnv_Variable_CfgVS2019X64"] !== "" && env["MyEnv_Variable_CfgVS2019X64"] !== undefined) {
                log.warning(`ENV["MyEnv_Variable_CfgVS2019X64"] = "${env["MyEnv_Variable_CfgVS2019X64"]}"`);
             }

             if (env["MyEnv_Variable_BuildFromCfgVS2019X64"] !== "" && env["MyEnv_Variable_BuildFromCfgVS2019X64"] !== undefined) {
                log.warning(`ENV["MyEnv_Variable_BuildFromCfgVS2019X64"] = "${env["MyEnv_Variable_BuildFromCfgVS2019X64"]}"`);
             }

             if (env["MyEnv_Variable_TestFromCfgVS2019X64"] !== "" && env["MyEnv_Variable_TestFromCfgVS2019X64"] !== undefined) {
                log.warning(`ENV["MyEnv_Variable_TestFromCfgVS2019X64"] = "${env["MyEnv_Variable_TestFromCfgVS2019X64"]}"`);
             }
         }
      }

     return newEnv;
  }

```
I select the following kit:
```
  {
    "name": "Visual Studio Enterprise 2019 Release - x86",
    "visualStudio": "c5421408",
    "visualStudioArchitecture": "x86",
    "environmentVariables": {
      "MyEnv_Variable_Kit": "MyEnv_Value_Kit"
    },
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "win32",
      "toolset": "host=x86"
    }
  },

```
Or the following kit:
```
  {
    "name": "Visual Studio Enterprise 2019 Release - amd64",
    "visualStudio": "c5421408",
    "visualStudioArchitecture": "x64",
    "environmentSetupScript": "c:\\temp\\subdir\\script.bat",
    "preferredGenerator": {
      "name": "Visual Studio 16 2019",
      "platform": "x64",
      "toolset": "host=x64"
    }
  },

```
while having this setup script:

```
echo Script echoing something ...
set MyEnv_Variable_Script=MyEnv_Value_Script

```

Maybe I have a wrong assumption in how I construct this artificial abstract repro?

**Comment by lygstate** [comment]
Hi, I investigated this, this is not a regression of my PR, I'll create a new PR for this issue to demostrate it. Ineed, when I revise the environment variables codes, I am confusing with the code in `effectiveKitEnvironment`

**Comment by elahehrashedi** [comment]
opened a new issue on this https://github.com/microsoft/vscode-cmake-tools/issues/2364

**Comment by bobbrow** [comment]
The fix for this issue is available in 1.10 "pre-release".  You can help us validate it by:
1. Opening the extensions panel in VS Code
2. Selecting the CMake Tools extension
3. Clicking the "Switch to Pre-release Version" button

![image](https://user-images.githubusercontent.com/12818240/156218538-7bce4a1f-4fc9-43cc-afb0-7f423aa5980e.png)
